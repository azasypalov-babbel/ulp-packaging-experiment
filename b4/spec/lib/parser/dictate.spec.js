import fixture from '../../fixtures/unparsed/dictate.json';
import parseDictate from '../../../src/lib/parser/dictate';

describe('Parser', () => {
  describe('Dictate', () => {
    const parsed = parseDictate(fixture);

    test('maps the data onto a different format', () => {
      expect(parsed).toMatchSnapshot();
    });

    describe('when "interaction" is "choose"', () => {
      describe('and "dictate" is "true"', () => {
        const chooseDictateDictate = Object.assign({}, fixture, {
          interaction: 'choose',
          dictate: true
        });

        test('maps "interaction" to "ChoiceButtons"', () => {
          expect(parseDictate(chooseDictateDictate).interaction).toBe('ChoiceButtons');
        });
      });
    });

    describe('when "interaction" is "write"', () => {
      describe('and "dictate" is "true"', () => {
        const writeDictateDictate = Object.assign({}, fixture, {
          interaction: 'write',
          dictate: true
        });

        test('maps "interaction" to "Fillin"', () => {
          expect(parseDictate(writeDictateDictate).interaction).toBe('Fillin');
        });

        describe('and "puzzle_helper" is "true"', () => {
          const writeDictatePuzzleHelperDictate = Object.assign({}, fixture, {
            interaction: 'write',
            dictate: true,
            puzzle_helper: true
          });

          test('maps "interaction" to "PuzzleHelper"', () => {
            expect(parseDictate(writeDictatePuzzleHelperDictate).interaction).toBe('PuzzleHelper');
          });
        });
      });
    });

    describe('when "interaction" is "sort"', () => {
      describe('and "dictate" is "true"', () => {
        const sortDictateDictate = Object.assign({}, fixture, {
          interaction: 'sort',
          dictate: true
        });

        test('maps "interaction" to "WordOrder"', () => {
          expect(parseDictate(sortDictateDictate).interaction).toBe('WordOrder');
        });
      });
    });

    describe('when "translation_visibility" is "full"', () => {
      const fullTranslationDialog = Object.assign({}, fixture, {
        translation_visibility: 'full'
      });

      test('maps "translation" to "Show"', () => {
        expect(parseDictate(fullTranslationDialog).translation).toBe('Show');
      });
    });

    describe('when "translation_visibility" is "partial"', () => {
      const partialTranslationDialog = Object.assign({}, fixture, {
        translation_visibility: 'partial'
      });

      test('maps "translation" to "Icon"', () => {
        expect(parseDictate(partialTranslationDialog).translation).toBe('Icon');
      });
    });

    describe('when "translation_visibility" is "none"', () => {
      const noneTranslationDialog = Object.assign({}, fixture, {
        translation_visibility: 'none'
      });

      test('maps "translation" to "Hide"', () => {
        expect(parseDictate(noneTranslationDialog).translation).toBe('Hide');
      });
    });
  });
});
