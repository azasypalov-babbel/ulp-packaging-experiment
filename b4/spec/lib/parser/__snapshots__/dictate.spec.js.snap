// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parser Dictate maps the data onto a different format 1`] = `
Object {
  "component": "dictate",
  "interaction": "ChoiceButtons",
  "mode": "Speak",
  "traineritems": Array [
    Object {
      "id": 0,
      "image_id": null,
      "info_text": "",
      "l1_text": "Ã‰coutez et choisissez la bonne lettre",
      "l2_text": "",
      "sound_id": null,
      "speaker_role": null,
      "type": "title",
    },
    Object {
      "id": "89290b8f9583d1c5c092dd3c4794580a",
      "image_id": null,
      "info_text": "",
      "l1_text": "f",
      "l2_text": "((*f|w))",
      "sound_id": "da96f13f2758241448cd8c4fbc543ef4",
      "speaker_role": "M1",
      "type": "Phrase",
    },
    Object {
      "id": "8e9741ed258b2fad8d7371d199ad986b",
      "image_id": null,
      "info_text": "",
      "l1_text": "h",
      "l2_text": "((a|*h))",
      "sound_id": "63eb1bac7dda623f4ba49976676ed0c6",
      "speaker_role": "F1",
      "type": "Phrase",
    },
  ],
  "translation": "Show",
}
`;
